drivers: ieee802154_nrf5: Align shim layer to 64-bit time

e-rk opened the pull request:
The updated nrf_802154 API accepts 64-bit time in microseconds.
The shim layer is updated to use 64-bit time.

Signed-off-by: Rafał Kuźnia <rafal.kuznia@nordicsemi.no>

To be done:

- [x] hal_nordic update

e-rk committed:
drivers: ieee802154_nrf5: Align shim layer to 64-bit time

The updated nrf_802154 API accepts 64-bit time in microseconds.
The shim layer is updated to use 64-bit time.

Signed-off-by: Rafał Kuźnia <rafal.kuznia@nordicsemi.no>

e-rk commented:
Pinging @jciupis @ankuns

ankuns commented on drivers/ieee802154/ieee802154_nrf5.c:
```diff
@@ -734,9 +733,83 @@ static void nrf5_config_mac_keys(struct ieee802154_key *mac_keys)
+/**
+ * @brief Convert 32-bit target time to absolute 64-bit target time.
+ */
+uint64_t target_time_convert_to_64_bits(uint32_t target_time)
```
shouldn't it be `static` ?

e-rk replied:
Fixed

e-rk committed:
manifest: update hal_nordic revision

Updated hal_nordic module revision to bring latest changes to
nrf_802154 component.

Signed-off-by: Rafał Kuźnia <rafal.kuznia@nordicsemi.no>

e-rk committed:
drivers: timer: Increase user channel count to 3 on nRF52

Increase the default user-allocable number of RTC channels to 3,
which is the numer of physical RTC CC channels not used by Zephyr
on nRF52 series SoCs.

Signed-off-by: Rafał Kuźnia <rafal.kuznia@nordicsemi.no>

edmont commented on drivers/ieee802154/ieee802154_nrf5.c:
```diff
@@ -473,12 +473,11 @@ static bool nrf5_tx_at(struct net_pkt *pkt, uint8_t *payload, bool cca)
 	bool ret;
 
 	ret = nrf_802154_transmit_raw_at(payload,
-					 tx_at - TXTIME_OFFSET_US,
```
Should be safe to remove the `TXTIME_OFFSET_US` definition above as well.

e-rk replied:
Removed the definition.

rlubos commented on west.yml:
```diff
@@ -88,7 +88,7 @@ manifest:
         - hal
     - name: hal_nordic
-      revision: eab07c9d86341b9f249afe8188cceab3b0dc90ef
+      revision: pull/111/head
```
Please update SHA: `1f9145e8c8359f103b4ac5085c6f5d2e66a2557d`

e-rk replied:
Done

anangl commented on drivers/ieee802154/ieee802154_nrf5.c:
```diff
@@ -734,9 +732,83 @@ static void nrf5_config_mac_keys(struct ieee802154_key *mac_keys)
+	 */
+	uint64_t now_us = nrf_802154_time_get();
+	uint32_t now_us_wrapped = (uint32_t)now_us;
+	uint32_t target_time_us_wrapped = (uint32_t)target_time;
```
Is this needed? `target_time` is already an `uint32_t`.

e-rk replied:
Removed the variable.

anangl commented on drivers/ieee802154/ieee802154_nrf5.c:
```diff
@@ -734,9 +732,83 @@ static void nrf5_config_mac_keys(struct ieee802154_key *mac_keys)
+			 * Subtract a 32-bit overflow and replace the least significant
+			 * 32 bits with the provided target time.
+			 */
+			result = now_us - (UINT32_MAX + 1);
```
Since `UINT32_MAX` is a 32-bit unsigned integer, `(UINT32_MAX + 1)` will get wrapped to 0.
You need to use `now_us - UINT32_MAX - 1` or typecast `UINT32_MAX` to `uint64_t`.

e-rk replied:
Changed to `now_us - UINT32_MAX - 1`

e-rk commented:
Rebased due to CI failure with unknown reason.
